lucille-core/src/test/java/com/kmwllc/lucille/core/WorkerTest.java

```diff
@@ -0,0 +1,70 @@
+package com.kmwllc.lucille.core;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.Test;
+import org.mockito.MockedConstruction;
+import org.mockito.MockedStatic;
+import org.mockito.Mockito;
+import com.kmwllc.lucille.message.WorkerMessenger;
+import com.kmwllc.lucille.message.WorkerMessengerFactory;
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigFactory;
+
+public class WorkerTest {
+
+  @Test
+  public void testMain() throws Exception {
+    Map<WorkerPool, List<Object>> args = new HashMap<>();
+    Config config = ConfigFactory.load("WorkerTest/config.conf");
+    WorkerMessengerFactory mockFactory = new WorkerMessengerFactory() {
+      @Override
+      public WorkerMessenger create() {
+        throw new UnsupportedOperationException("Unimplemented method 'create'");
+      }
+    };
+    WorkerMessengerFactory mockFactory2 = new WorkerMessengerFactory() {
+      @Override
+      public WorkerMessenger create() {
+        throw new UnsupportedOperationException("Unimplemented method 'create'");
+      }
+    };
+    try (MockedStatic<ConfigFactory> configFactory = Mockito.mockStatic(ConfigFactory.class);
+        MockedConstruction<WorkerPool> workerPool = Mockito.mockConstruction(WorkerPool.class, (mock, context) -> {
+          args.put(mock, new ArrayList<>(context.arguments()));
+        });
+        MockedStatic<WorkerMessengerFactory> factory = Mockito.mockStatic(WorkerMessengerFactory.class)) {
+
+      configFactory.when(ConfigFactory::load).thenReturn(config);
+
+      factory.when(() -> {
+        WorkerMessengerFactory.getKafkaFactory(config, "foo");
+      }).thenReturn(mockFactory);
+
+      factory.when(() -> {
+        WorkerMessengerFactory.getKafkaFactory(config, "foo2");
+      }).thenReturn(mockFactory2);
+
+      Worker.main(new String[0]);
```

Comment:
main() is meant as the entry point for launching Worker as a java app/process so it's highly unlikely that main() would ever be called twice in a row like this. I see that this is testing main() in two cases, with and without args. I think this would be more natural as two separate tests... testMainWithArgs() and testMainWithoutArgs()...